ArduCopter/mode_auto.cpp

```diff
@@ -335,19 +335,18 @@ void ModeAuto::takeoff_start(const Location& dest_loc)
         alt_target_terrain = true;
     } else {
         // set horizontal target
-        Location dest(dest_loc);
-        dest.lat = copter.current_loc.lat;
-        dest.lng = copter.current_loc.lng;
+        takeoff_loc = Location(copter.current_loc.lat, copter.current_loc.lng, dest_loc.alt, dest_loc.get_alt_frame());
```

Comment:
Can we have a comment to explain why we are putting the current location in.  It will help with preventing confusion in the future.